src/sentry/scim/endpoints/teams.py

```diff
@@ -192,6 +192,13 @@ def post(self, request: Request, organization) -> Response:
         request.data.update(
             {"name": request.data["displayName"], "slug": slugify(request.data["displayName"])}
         ),
+
+        if "scim" in request.user.username:
```

Comment:
This condition looks like it's too broad. I'm not 100% clear on the context, but I'd assume that any regular user could make this POST request, and their username might coincidentally contain the substring `"scim"`.

Reply:
All of the SCIM internal integrations have the form scim-internal-integration-{uuid} (and no email or name, I think). Should I check for that instead?

Reply:
Yes, that sounds good.

In a perfect system, I'd prefer to tell them apart without relying on user attributes at all. It might be worth looking into whether there is some model for SCIM internal integrations that points to these user IDs. It would be worth a database query to check it that way, in my opinion.

Otherwise, yes, checking with a `scim-internal-integration-(.*)` regular expression sounds good, and you can capture the ID part of the string at the same time. (Let me know if you'd like help with that.) It also makes sense to make the check as specific as possible, such as with the lack of an email and name as you say.